examples/fairmotion_interface_utils.py

```diff
@@ -0,0 +1,342 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import time
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Optional
+
+import magnum as mn
+from fairmotion.data import amass
+from fairmotion.ops import motion as motion_ops
+
+from habitat_sim.logging import LoggingContext, logger
+
+LoggingContext.reinitialize_from_env()
+
+#### Constants ###
+ROOT, FIRST, LAST = 0, 0, -1
+
+
+class MType(Enum):
+    TRANSITIVE = 0
+    SCENIC = 1
+
+
+class Motions:
+    """
+    The Move class is collection of stats that will hold the different movement motions
+    for the character to use when following a path. The character is left-footed so that
+    is our reference for with step the motions assume first.
+    """
+
+    @dataclass
+    class MotionData:
+        """
+        A class intended to handle precomputations of utilities of the motion we want to
+        load into the character.
+        """
+
+        # transitive motion is necessary for scenic motion build
+        def __init__(
+            self,
+            motion_,
+            type_,
+            transitive_motion_=None,
+            scenic_direction_offset: int = 0,
+        ) -> None:
+            logger.info("Loading Motion data...")
+            print("Loading Motion data...")  # the logger line above is not printing
+            if type_ == MType.TRANSITIVE:
+                # primary
+                self.motion = motion_
+                self.type = MType.TRANSITIVE
+                self.poses = motion_.poses
+                self.fps = motion_.fps
+                self.num_of_frames: int = len(motion_.poses)
+                self.map_of_total_displacement: float = []
+                self.center_of_root_drift: mn.Vector3 = mn.Vector3()
+                self.time_length = self.num_of_frames * (1.0 / motion_.fps)
+
+                # intermediates
+                self.translation_drifts: List[mn.Vector3] = []
+                self.forward_displacements: List[mn.Vector3] = []
+                self.root_orientations: List[mn.Quaternion] = []
+
+                # first and last frame root position vectors
+                f = motion_.poses[0].get_transform(ROOT, local=False)[0:3, 3]
+                f = mn.Vector3(f)
+                l = motion_.poses[LAST].get_transform(ROOT, local=False)[0:3, 3]
+                l = mn.Vector3(l)
+
+                # axis that motion uses for up and forward
+                self.direction_up = mn.Vector3.z_axis()
+                forward_V = (l - f) * (mn.Vector3(1.0, 1.0, 1.0) - mn.Vector3.z_axis())
```

Comment:
equivalent?
```suggestion
                forward_V = (l - f) * mn.Vector3(1.0, 1.0, 0.0)
```

Reply:
I wrote it like this to better illustrate exact what I am doing. Would you still like me to make your change?

Reply:
Fair enough. We can leave it.